src/MongoDB.Driver.Core/Core/Compression/Snappy/Native/SnappyNativeMethods.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            return _snappyNativeMethods.snappy_validate_compressed_buffer(input, input_length);
+        }
+
+        // nested types
+        private class SnappyNativeMethods32 : ISnappyNativeMethods
+        {
+            private readonly Delegates32.snappy_compress _snappy_compress;
+            private readonly Delegates32.snappy_max_compressed_length _snappy_max_compressed_length;
+            private readonly Delegates32.snappy_uncompress _snappy_uncompress;
+            private readonly Delegates32.snappy_uncompressed_length _snappy_uncompressed_length;
+            private readonly Delegates32.snappy_validate_compressed_buffer _snappy_validate_compressed_buffer;
+
+            public SnappyNativeMethods32(ILibraryLoaderSource libraryLoaderSource)
+            {
+                _snappy_compress = libraryLoaderSource.GetFunction<Delegates32.snappy_compress>(nameof(snappy_compress));
+                _snappy_max_compressed_length = libraryLoaderSource.GetFunction<Delegates32.snappy_max_compressed_length>(nameof(snappy_max_compressed_length));
+                _snappy_uncompress = libraryLoaderSource.GetFunction<Delegates32.snappy_uncompress>(nameof(snappy_uncompress));
+                _snappy_uncompressed_length = libraryLoaderSource.GetFunction<Delegates32.snappy_uncompressed_length>(nameof(snappy_uncompressed_length));
+                _snappy_validate_compressed_buffer = libraryLoaderSource.GetFunction<Delegates32.snappy_validate_compressed_buffer>(nameof(snappy_validate_compressed_buffer));
+            }
+
+            // public methods
+            public SnappyStatus snappy_compress(IntPtr input, int input_length, IntPtr output, ref int output_length)
+            {
+                return _snappy_compress(input, input_length, output, ref output_length);
+            }
+
+            public int snappy_max_compressed_length(int input_length)
+            {
+                return _snappy_max_compressed_length(input_length);
+            }
+
+            public SnappyStatus snappy_uncompress(IntPtr input, int input_length, IntPtr output, ref int output_length)
+            {
+                return _snappy_uncompress(input, input_length, output, ref output_length);
+            }
+
+            public SnappyStatus snappy_uncompressed_length(IntPtr input, int input_length, out int output_length)
+            {
+                return _snappy_uncompressed_length(input, input_length, out output_length);
+            }
+
+            public SnappyStatus snappy_validate_compressed_buffer(IntPtr input, int input_length)
+            {
+                return _snappy_validate_compressed_buffer(input, input_length);
+            }
+
+            // nested types
+            private class Delegates32
+            {
+                public delegate SnappyStatus snappy_compress(IntPtr input, int input_length, IntPtr output, ref int output_length);
```

Comment:
nit - use uint32 (uint) since size_t is unsigned.

Reply:
done